libflux/src/flux/semantic/builtins.rs

```diff
@@ -275,6 +275,267 @@ pub fn builtins() -> Builtins<'static> {
+                "limit"  => Node::Builtin("forall [t0] (<-tables: [t0], n: int, ?offset: int) -> [t0]"),
+                "linearBins" => Node::Builtin(r#"
+                    forall [] (
+                        start: float,
+                        width: float,
+                        count: int,
+                        ?infinity: bool
+                    ) -> [float]
+                "#),
+                "logarithmicBins" => Node::Builtin(r#"
+                    forall [] (
+                        start: float,
+                        factor: float,
+                        count: int,
+                        ?infinity: bool
+                    ) -> [float]
+                "#),
+                "map" => Node::Builtin("forall [t0, t1] (<-tables: [t0], fn: (r: t0) -> t1, ?mergeKey: bool) -> [t1]"),
+                "max" => Node::Builtin("forall [t0] where t0: Row (<-tables: [t0], ?column: string) -> [t0]"),
+                "mean" => Node::Builtin(r#"
+                    forall [t0, t1] where t0: Row, t1: Row (
+                        <-tables: [t0],
+                        ?column: string
+                    ) -> [t1]
+                "#),
+                "min" => Node::Builtin("forall [t0] where t0: Row (<-tables: [t0], ?column: string) -> [t0]"),
+                "mode" => Node::Builtin(r#"
+                    forall [t0, t1, t2] where t0: Row, t2: Row (
+                        <-tables: [t0],
+                        ?column: string
+                    ) -> [{_value: t1 | t2}]
+                "#),
+                "movingAverage" => Node::Builtin("forall [t0, t1] where t0: Numeric (<-tables: [{_value: t0 | t1}], n: int) -> [{_value: float | t1}]"),
+                "pivot" => Node::Builtin(r#"
+                    forall [t0, t1] where t0: Row, t1: Row (
+                        <-tables: [t0],
+                        rowKey: [string],
+                        columnKey: [string],
+                        valueColumn: string
+                    ) -> [t1]
+                "#),
+                "quantile" => Node::Builtin(r#"
+                    forall [t0] where t0: Row (
+                        <-tables: [t0],
+                        ?column: string,
+                        q: float,
+                        ?compression: float,
+                        ?method: string
+                    ) -> [t0]
+                "#),
+                // start and stop should be able to constrained to time or duration with a kind constraint:
+                // https://github.com/influxdata/flux/issues/2243
+                "range" => Node::Builtin(r#"
+                    forall [t0, t1] where t0: Row, t3: Row (
+                        <-tables: [t0],
+                        start: t1,
+                        ?stop: t2,
+                        ?timeColumn: string,
+                        ?startColumn: string,
+                        ?stopColumn: string
+                    ) -> [t3]
+                "#),
```

Comment:
`range` should add `_start` and `_stop` labels to the output rows. We should create an issue to remove the parameters `timeColumn`, `startColumn`, and `stopColumn`.

Reply:
Ok, I'll file an issue and mention it here in a comment.